python-lib/dku_deeplearning_image/recipes/retrain_recipe.py

```diff
@@ -118,36 +116,60 @@ def _retrain(self, train_generator, test_generator, callback_list):
             verbose=2
         )
 
+    def _get_augmented_images(self, image, extra_images_gen):
+        def _run_image_augm(im, augm_gen):
+            image_augm = np.tile(im, (self.config.data_augmentation, 1, 1, 1))
+            return next(augm_gen.flow(image_augm, batch_size=self.config.data_augmentation))
+
+        return tf.numpy_function(
+            func=lambda x: _run_image_augm(x, extra_images_gen),
+            inp=[image],
+            Tout=tf.float32)
+
+    def _add_data_augmentation(self, X_tfds, y_values):
+        extra_images_gen = self._get_tf_image_data_gen()
+        X_tfds = X_tfds.map(map_func=lambda x: self._get_augmented_images(x, extra_images_gen),
+                            num_parallel_calls=constants.AUTOTUNE)
+        X_tfds = X_tfds.flat_map(map_func=lambda x: tf.data.Dataset.from_tensor_slices(x))
+        y_values = np.repeat(y_values, self.config.n_augmentation, axis=0)
+        return X_tfds, y_values
+
+    def _build_tfds(self, pddf, images_folder, ignore_augm=False):
```

Comment:
`pddf` is a bit cryptic, `df` would be enough no ?

Reply:
Fixed in 8806f43